src/java/org/apache/cassandra/transport/ExceptionHandlers.java

```diff
@@ -133,6 +136,15 @@ else if (Throwables.anyCauseMatches(cause, t -> t instanceof Errors.NativeIoExce
             ClientMetrics.instance.markUnknownException();
             logger.trace("Native exception in client networking", cause);
         }
+        else if (Throwables.anyCauseMatches(cause, t -> t instanceof SSLHandshakeException))
+        {
+            ClientMetrics.instance.markSSLHandshakeException();
+            NoSpamLogger.log(logger, NoSpamLogger.Level.WARN, 1, TimeUnit.MINUTES, "SSLHandshakeException in client networking with peer {} {}", clientAddress, cause.getMessage());
+        }
+        else if (Throwables.anyCauseMatches(cause, t -> t instanceof SSLException))
+        {
+            NoSpamLogger.log(logger, NoSpamLogger.Level.WARN, 1, TimeUnit.MINUTES, "SSLException in client networking with peer {} {}", clientAddress, cause.getMessage());
+        }
```

Comment:
The only thing that worries me a little bit here is what happens when we have overloads, protocol exceptions, and SSL problems all occurring at the same time. It's not that expensive to create a new logger to wrap, specifically for SSL exceptions. Any objection to doing that?

Reply:
I am not sure I am getting your comment right ... The way how NoSpamLogger seems to work is that it will not repeat same message all over again during some period. It does not mean it will not repeat any message if some was already logged.

Example: lets say we have messages to log, "msg1" and "msg2". What NoSpamLogger does is that it will log "msg1" and then "msg2" but if you ever send any other "msg2" to it during 1 minute, it will not log that. But that does not mean that it will not log msg1 if 1 minute passed from its last log. 

In other words, these periods of 1 minute are _per message_, not _per logger_. Logger via which we log these messages might be same.

Reply:
You are 100% correct. I only looked at `NoSpamLogger` acquisition and ignored `NoSpamLogStatement` acquisition.